examples/nlp/language_modeling/megatron_gpt_continue_training.py

```diff
@@ -1,198 +0,0 @@
-        gpt_cfg.optim = cfg.model.optim
-        gpt_cfg.precision = cfg.trainer.precision
-        gpt_cfg.restore_from_path = cfg.restore_from_path
-        gpt_cfg.resume_from_checkpoint = cfg.model.resume_from_checkpoint
-        gpt_cfg.gradient_as_bucket_view = cfg.model.gradient_as_bucket_view
-        gpt_cfg.encoder_seq_length = cfg.model.encoder_seq_length
-        gpt_cfg.max_position_embeddings = cfg.model.max_position_embeddings
-        gpt_cfg.seq_len_interpolation_factor = cfg.model.seq_len_interpolation_factor
-        gpt_cfg.use_flash_attention = cfg.model.use_flash_attention
-        gpt_cfg.tensor_model_parallel_size = cfg.model.get('tensor_model_parallel_size', 1)
-        gpt_cfg.pipeline_model_parallel_size = cfg.model.get('pipeline_model_parallel_size', 1)
-        gpt_cfg.pipeline_model_parallel_split_rank = cfg.model.get('pipeline_model_parallel_split_rank', 0)
-
-        # This is needed when modifying a hparam file directly to load `.ckpt` files.
-        # This is not needed to modify the cfg in `.nemo` files.
-        if add_cfg_to_tree:
-            OmegaConf.resolve(gpt_cfg)
-            gpt_cfg.cfg = gpt_cfg
-
-    return gpt_cfg
-
-
-def load_from_nemo(cls, cfg, trainer, gpt_cfg, modify_confg_fn):
-    gpt_cfg = modify_confg_fn(gpt_cfg, cfg, add_cfg_to_tree=False)
-    save_restore_connector = NLPSaveRestoreConnector()
-    if os.path.isdir(cfg.restore_from_path):
-        save_restore_connector.model_extracted_dir = cfg.restore_from_path
-    model = cls.restore_from(
-        restore_path=cfg.restore_from_path,
-        trainer=trainer,
-        override_config_path=gpt_cfg,
-        save_restore_connector=save_restore_connector,
-    )
-    return model
-
-
-def load_from_checkpoint_dir(cls, cfg, trainer, modify_confg_fn):
-    app_state = AppState()
-    if cfg.model.tensor_model_parallel_size > 1 or cfg.model.pipeline_model_parallel_size > 1:
-        app_state.model_parallel_size = cfg.model.tensor_model_parallel_size * cfg.model.pipeline_model_parallel_size
-        app_state.tensor_model_parallel_size = cfg.model.tensor_model_parallel_size
-        app_state.pipeline_model_parallel_size = cfg.model.pipeline_model_parallel_size
-        (
-            app_state.tensor_model_parallel_rank,
-            app_state.pipeline_model_parallel_rank,
-            app_state.model_parallel_size,
-            app_state.data_parallel_size,
-            app_state.pipeline_model_parallel_split_rank,
-            app_state.virtual_pipeline_model_parallel_rank,
-        ) = fake_initialize_model_parallel(
-            world_size=app_state.model_parallel_size,
-            rank=trainer.global_rank,
-            tensor_model_parallel_size_=cfg.model.tensor_model_parallel_size,
-            pipeline_model_parallel_size_=cfg.model.pipeline_model_parallel_size,
-            pipeline_model_parallel_split_rank_=cfg.model.pipeline_model_parallel_split_rank,
-        )
-    checkpoint_path = inject_model_parallel_rank(
```

Comment:
Wondering about the legacy tp_pp partition checkpoint. With this refactor, I don't see a clear way to resume from a checkpoint in this format if we do not save the .nemo file?

```
checkpoint/
  tp_rank_00_pp_rank_000/
    megatron_gpt--....-consumed_samples=1000.ckpt
  tp_rank_00_pp_rank_001/
  ...
  tp_rank_01_pp_rank_000/
  tp_rank_01_pp_rank_001/
  ...
```

Reply:
@ryxli it will still work. It uses the lightning resume for ckpt checkpoints (the logic is in NLPoverride), so both legacy and dist format will work. This part of code is redundant anyways.
https://github.com/NVIDIA/NeMo/blob/bbbe826aa524d2ef3d841e50ec97788322a31f4e/nemo/collections/nlp/parts/nlp_overrides.py#L521